packages/clarity-js/src/data/metadata.ts

```diff
@@ -108,18 +100,33 @@ function tab(): string {
 export function save(): void {
   let ts = Math.round(Date.now());
   let upgrade = config.lean ? BooleanFlag.False : BooleanFlag.True;
-  let upload = typeof config.upload === Constant.String ? config.upload : Constant.Empty;
-  let server = config.server ? config.server.replace(Constant.HTTPS, Constant.Empty) : Constant.Empty;
+  let upload = config.upload && typeof config.upload === Constant.String ? config.upload as string : Constant.Empty;
+  let host: string = Constant.Empty;
+  let path: string = Constant.Empty;
+
+  // The code below checks if the "upload" value is a string, and if so, break it into "host" and "path" before writing to session cookie
+  // This is for forward compatibility - to be removed in future versions (v0.6.21)
+  if (upload) {
+    host = upload.substr(0, upload.indexOf("/", Constant.HTTPS.length)); // Look for first "/" starting after initial "https://" string
+    path = host.length > 0 && host.length < upload.length ? upload.substr(host.length + 1) : upload; // Grab path of the url and update host value
+    host = host.replace(Constant.HTTPS, Constant.Empty);
+  }
   if (upgrade && callback) { callback(data, !config.lean); }
-  setCookie(Constant.SessionKey, [data.sessionId, ts, data.pageNum, upgrade, upload, server].join(Constant.Pipe), Setting.SessionExpire);
+  setCookie(Constant.SessionKey, [data.sessionId, ts, data.pageNum, upgrade, path, host].join(Constant.Pipe), Setting.SessionExpire);
 }
 
 function supported(target: Window | Document, api: string): boolean {
   try { return !!target[api]; } catch { return false; }
 }
 
-function track(): void {
-  setCookie(Constant.CookieKey, `${data.userId}${Constant.Pipe}${Setting.CookieVersion}`, Setting.Expire);
+function track(expiry: number): void {
+  // Convert time precision into days to reduce number of bytes we have to write in a cookie
+  // E.g. 1628735962643 / (24*60*60*1000) => 18851 (days) => ejn in base36 (13 bytes => 3 bytes)
+  let end = Math.floor((Date.now() + (Setting.Expire * Time.Day))/Time.Day);
+  // To avoid cookie churn, write user id cookie only once every day
+  if (expiry === null || Math.abs(end - expiry) >= Setting.CookieInterval) {
+    setCookie(Constant.CookieKey, [data.userId, Setting.CookieVersion, end.toString(36)].join(Constant.Pipe), Setting.Expire);
```

Comment:
>CookieVersion [](http://example.com/codeflow?start=56&length=13)

Wouldn't this break the cookie decoding for data collection. Should we increment the cookie version?

Reply:
No it won't - this is a first party cookie and available on client only. I was planning to upgrade version for breaking changes only.